src/cli/add.rs

```diff
@@ -412,6 +413,11 @@ fn update_pypi_specs_from_lock_file(
                 })
                 .collect_vec()
                 .iter(),
+            project
+                .config()
+                .pinning_strategy
+                .clone()
+                .unwrap_or(PinningStrategy::Semver),
```

Comment:
This seems like duplication. Maybe we can compute it outside the loops?

Reply:
Removed